applications/RANSApplication/python_scripts/formulations/utilities.py

```diff
@@ -0,0 +1,244 @@
+                                             move_mesh_flag):
+    return newton_raphson_strategy(model_part, scheme,
+                                   convergence_criteria, builder_and_solver,
+                                   max_iterations, compute_reactions,
+                                   reform_dofs_at_each_step, move_mesh_flag)
+
+
+def CreateLinearSolver(solver_settings):
+    return linear_solver_factory.ConstructSolver(solver_settings)
+
+
+def CreateResidualCriteria(variable_convergence_criteria_info_list):
+    return residual_criteria(variable_convergence_criteria_info_list)
+
+
+def CreateIncrementalUpdateScheme():
+    return incremental_update_static_scheme()
+
+
+def CalculateNormalsOnConditions(model_part):
+    domain_size = model_part.ProcessInfo[Kratos.DOMAIN_SIZE]
+    if (not RansVariableUtilities.IsAnalysisStepCompleted(
+            model_part, "CONDITION_NORMAL_CALCULATION")):
+
+        # this calculates normals on whole model part, and assigns
+        # NORMAL variable in NodalSolutionStepDataValue container.
+        # NORMAL on conditions is required by some formulations such as inlet condition for
+        # incompressible potential flow velocity formulation, and all boundaries for incompressible
+        # potential flow pressure formulation.
+        Kratos.NormalCalculationUtils().CalculateOnSimplex(
+            model_part.Conditions, domain_size)
+        RansVariableUtilities.AddAnalysisStep(model_part,
+                                              "CONDITION_NORMAL_CALCULATION")
+
+        # This reverts incorrectly calculated nodal NORMALS from previous method
+        # since it spreads condition NORMAL to all nodes of model part, but from this
+        # method, it again spreads condition NORMALs to nodes where condition is applied
+        # with SLIP flag.
+        RansVariableUtilities.AssignConditionVariableValuesToNodes(
+            model_part, Kratos.NORMAL, Kratos.SLIP)
+
+        Kratos.Logger.PrintInfo("NormalCalculationUtils",
+                                "Condition normals calculated.")
+
+
+def CreateRansFormulationModelPart(formulation, element_name, condition_name):
+    formulation.domain_size = formulation.GetBaseModelPart().ProcessInfo[
+        Kratos.DOMAIN_SIZE]
+
+    element_suffix = str(
+        formulation.domain_size) + "D" + str(formulation.domain_size + 1) + "N"
+    condition_suffix = str(formulation.domain_size) + "D" + str(
+        formulation.domain_size) + "N"
+
+    element_name = element_name + element_suffix
+    condition_name = condition_name + condition_suffix
+    return CreateDuplicateModelPart(
+        formulation.GetBaseModelPart(),
+        formulation.GetName() + "_" + element_name + "_" + condition_name,
+        element_name, condition_name, "")
+
+
+def IsBufferInitialized(formulation):
+    return (formulation.GetBaseModelPart().ProcessInfo[Kratos.STEP] + 1 >=
+            formulation.GetMinimumBufferSize())
```

Comment:
Shouldn't this be a method of the `Formulation` class?

Reply:
done :)